### ISSUE TITLE
eventTypes.join is not a function

### ISSUE THREAD

**Issue by alexdorandmec** [matt-filion/serverless-external-s3-event#67]
When trying to run the following command:

```
 sls s3deploy
```

I get the following exception: 
```
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command s3deploy
Serverless: Invoke s3deploy
Serverless: beforeFunctions --> building ... 
Serverless: beforeFunctions <-- Complete, built 1 events.
Serverless: functions --> prepare to be executed by s3 buckets ... 
policyId exS3-v2-d365-int-dev-VendorExport-d365-import-bucket-501695966922
Serverless: functions <-- built 0 events across 1 buckets. 
Serverless: beforeS3 --> 
Serverless: beforeS3 <-- 
Serverless: s3 --> initiate requests ...
 
  Type Error ---------------------------------------------
 
  eventTypes.join is not a function
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Stack Trace --------------------------------------------
 
TypeError: eventTypes.join is not a function
    at S3.getId (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/S3.js:15:51)
    at BucketConfig.isNew (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/BucketConfig.js:32:22)
    at fileConfig.events.forEach (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/BucketConfig.js:27:16)
    at Array.forEach (<anonymous>)
    at BucketConfig.addNewNotifications (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/BucketConfig.js:26:23)
    at BucketConfig.update (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/BucketConfig.js:21:10)
    at promises.bucketNotifications.filter.map.bucketConfigurationFromFile (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/index.js:152:24)
    at Array.map (<anonymous>)
    at S3Deploy.s3 (/Users/aldor/Projects/MS/erp-integration/functions/d365-dmf-recurring-api-export-component/node_modules/serverless-plugin-existing-s3/index.js:146:10)
    at BbPromise.reduce (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:390:55)
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:390:22)
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:421:17)
    at variables.populateService.then.then (/usr/local/lib/node_modules/serverless/lib/Serverless.js:157:33)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
From previous event:
    at Serverless.run (/usr/local/lib/node_modules/serverless/lib/Serverless.js:144:8)
    at serverless.init.then (/usr/local/lib/node_modules/serverless/bin/serverless:43:50)
    at <anonymous>
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           8.11.3
     Serverless Version:     1.32.0

```

Here is my serverless.yml : 

```
service: d365-int # NOTE: update this with your service name

custom:
  global:
    bucketName:
      Fn::Join:
      - ''
      - - 'd365-integration-'
        - ${self:provider.stage}
        - '-'
        - Ref: AWS::AccountId
    failedBucketName:
      Fn::Join:
      - ''
      - - 'd365-integration-failed-'
        - ${self:provider.stage}
        - '-'
        - Ref: AWS::AccountId
    completedBucketName:
      Fn::Join:
      - ''
      - - 'd365-integration-completed-'
        - ${self:provider.stage}
        - '-'
        - Ref: AWS::AccountId

  stage: ${opt:stage, self:provider.stage}
  region: ${self:provider.region}
  prefix: ${self:custom.stage}-${self:service}



provider:
  name: aws
  runtime: java8

  # you can overwrite defaults here
  stage: dev
  region: ca-central-1

  # you can add statements to the Lambda function's IAM Role here
  iamRoleStatements:
    - Effect: "Allow"
      Action:
      - "s3:*"
      - "s3:PutObject"
      - "s3:DeleteObject"
      - "s3:GetObjectTagging"
      - "s3:PutObjectTagging"
      - "s3:PutBucketNotification"
      Resource: "arn:aws:s3:::*"

  environment:
    oAuthUrl: "https://login.windows.net/microsofttest/oauth2/token"
    clientId: "12345678"
    clientSecret: "87654321"
    D365ServerUrl: "https://microsofttest.cloudax.dynamics.com"

    recurringDequeue: "/api/connector/dequeue/"
    recurringAck: "/api/connector/ack/"

    region: ${self:custom.region}
    accountId: aid:#{AWS::AccountId}
    bucketName: ${self:custom.global.bucketName}
    failedBucketName: ${self:custom.global.failedBucketName}
    completedBucketName: ${self:custom.global.completedBucketName}


# you can define service wide environment variables here
#  environment:
#    variable1: value1

# you can add packaging information here
package:
  artifact: target/d365-dmf-recurring-api-export-component-dev.jar

functions:
  VendorExport:
    handler: com.microsoft.d365.ProcessExportFromS3Handler
    events:
    - existingS3:
        # Known bug, we are hardcoding the trigger for now: https://github.com/matt-filion/serverless-external-s3-event/issues/66
        # bucket: ${self:custom.global.bucketName}
        bucket: "d365-import-bucket-501695966922"
        events: s3:ObjectCreated:*
    environment:
      name: "1B4D7CCD-CE10-4B97-B2D4-FFA978E40464"
      region: ${self:provider.region}

  ExportTrigger:
    handler: com.microsoft.d365.RecurringExportTriggerHandler
    events:
        - schedule:
            rate: rate(10 minutes)
            integration: "Vendor"

    #    Define function environment variables here
    environment:
      name: "1C4D7CCD-CE10-4B97-B2D4-FFA978E41464"
      region: ${self:provider.region}

# you can add CloudFormation resource templates here
resources:
  Resources:
    BucketName:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: ${self:custom.global.bucketName}
    FaileBucketName:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: ${self:custom.global.failedBucketName}
    CompletedBucketName:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: ${self:custom.global.completedBucketName}

plugins:
  - serverless-plugin-existing-s3
  - serverless-pseudo-parameters

```

**Comment by alexdorandmec** [comment]
Node version is v8.11.3

**Comment by valecarlos** [comment]
When I try that new version I'm getting:
```
An error occurred: RenameLambdaFunction - Value nodejs8.11 at 'runtime' failed to satisfy constraint: Member must satisfy enum value set: [java8, nodejs6.10, nodejs8.10, python2.7, python3.6, dotnetcore1.0, dotnetcore2.0, dotnetcore2.1, go1.x] or be a valid ARN (Service: AWSLambda; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: de6ba877-dc4d-11e8-af1c-25a8844cdb06).
```

**Comment by alexdorandmec** [comment]
It stopped working again. So Would you be kind and give a proper guide as to how the indentation should be? an example? (preferably with comment explaining number of spaces?)

**Comment by tamtamazatec** [comment]
```
functions:
  rename:
    handler: src/index.rename
    events:
      - existingS3:
          bucket: testbucket
          events: s3:ObjectCreated:*
          rules:
            - suffix: .txt
```
is wrong, correct way is
```
functions:
  rename:
    handler: src/index.rename
    events:
      - existingS3:
          bucket: testbucket
          events:
            - s3:ObjectCreated:*
          rules:
            - suffix: .txt
```

**Comment by matt-filion** [comment]
Fix pusehd to 2.3.2 to address this YML varaition (of a single value).
